src/openforms/forms/tests/admin/test_views.py

```diff
@@ -305,3 +308,271 @@ def test_single_import(self, m_import):
             messages[0].message,
         )
+
+
+@disable_admin_mfa()
+class TestMigrationForm(WebTest):
+    csrf_checks = False
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.admin_user = StaffUserFactory(user_permissions=["forms.change_form"])
+
+    def test_not_superuser_cant_access(self):
+        user = UserFactory(is_staff=False, is_superuser=False)
+        ogone_merchant = OgoneMerchantFactory()
+        form_to_migrate = FormFactory(
+            name="Ogone form 1",
+            payment_backend="ogone-legacy",
+            payment_backend_options={"merchant_id": ogone_merchant.pk},
+        )
+
+        redirect = self.app.post(
+            reverse("admin:forms_payment_migration"),
+            {"forms_to_migrate": [form_to_migrate.pk]},
+            user=user,
+        )
+
+        redirect_url = (
+            f"{reverse('admin:login')}?next={reverse('admin:forms_payment_migration')}"
+        )
+        self.assertRedirects(
+            redirect,
+            redirect_url,
+            fetch_redirect_response=False,
+            target_status_code=302,
+        )
+
+        form_to_migrate.refresh_from_db()
+
+        self.assertEqual(form_to_migrate.payment_backend, "ogone-legacy")
+        self.assertEqual(
+            form_to_migrate.payment_backend_options, {"merchant_id": ogone_merchant.pk}
+        )
+
+    def test_form_permissions(self):
+        user = StaffUserFactory(user_permissions=[])
+        ogone_merchant = OgoneMerchantFactory()
+        form_to_migrate = FormFactory(
+            name="Ogone form 1",
+            payment_backend="ogone-legacy",
+            payment_backend_options={"merchant_id": ogone_merchant.pk},
+        )
+
+        response = self.app.post(
+            reverse("admin:forms_payment_migration"),
+            {"forms_to_migrate": [form_to_migrate.pk]},
+            user=user,
+            expect_errors=True,
+        )
+
+        self.assertEqual(response.status_code, 403)
+
+        form_to_migrate.refresh_from_db()
+
+        self.assertEqual(form_to_migrate.payment_backend, "ogone-legacy")
+        self.assertEqual(
+            form_to_migrate.payment_backend_options, {"merchant_id": ogone_merchant.pk}
+        )
+
+    def test_confirmation_page(self):
+        ogone_merchant = OgoneMerchantFactory(
+            label="Merchant Y", pspid="merchant-y", endpoint_preset=OgoneEndpoints.test
+        )
+        FormFactory(
+            name="Ogone form 1",
+            payment_backend="ogone-legacy",
+            payment_backend_options={"merchant_id": ogone_merchant.pk},
+        )
+
+        response = self.app.post(
+            reverse("admin:forms_form_changelist"),
+            {
+                "action": "migrate_to_worldline",
+                "select_across": "0",
+                "index": "0",
+                "_selected_action": "1",
+            },
+            user=self.admin_user,
+        )
```

Comment:
when you're using web-test, you don't want to be bothering with these kind of payloads. Instead, use webtest to:

1. Navigate to the form change list
2. Locate and select an option in the dropdown using `admin_page.forms[...]["action"].select(...)`
3. Similarly check the checkbox of the form to migrate
4. Use webtest to submit the form `admin_page.forms[...].submit()`

and then verify the result.

Webtest/admin tests should closely mimick the user actions in the browser without being full-blown playwright end-to-end tests